sql/atxs/atxs.go

```diff
@@ -569,27 +571,53 @@ func IterateAtxsData(
 		weight uint64,
 		base uint64,
 		height uint64,
+		nonce *types.VRFPostIndex,
+		isMalicious bool,
 	) bool,
 ) error {
 	_, err := db.Exec(
-		`select id, pubkey, epoch, coinbase, effective_num_units, base_tick_height, tick_count
-		from atxs where epoch between ?1 and ?2;`,
-		func(stmt *sql.Statement) {
-			stmt.BindInt64(1, int64(from.Uint32()))
-			stmt.BindInt64(2, int64(to.Uint32()))
-		},
+		`select
+                   a.id, a.pubkey, a.epoch, a.coinbase, a.effective_num_units,
+                   a.base_tick_height, a.tick_count,
+                   coalesce(a.nonce, (
+		     select a1.nonce
+		     from atxs a1
+		     where a1.pubkey = a.pubkey and a1.epoch < a.epoch and a1.nonce is not null
+		     order by a1.epoch desc
+		     limit 1
+		   )) as nonce,
```

Comment:
it is also possible to copy nonce from the previous atx on insertion, it will add very little data but completely avoid this subqueries, and also this bug https://github.com/spacemeshos/go-spacemesh/issues/5701